src/test/java/io/stargate/sgv2/jsonapi/api/v1/CreateCollectionIntegrationTest.java

```diff
@@ -83,6 +83,79 @@ class CreateCollection {
                 }
                 """;
 
+    String createCollectionWithAllowIndexing =
+        """
+                {
+                  "createCollection": {
+                    "name": "simple_collection_allow_indexing",
+                    "options" : {
+                      "vector" : {
+                        "size" : 5,
+                        "function" : "cosine"
+                      },
+                      "indexing" : {
+                        "allow" : ["field1", "field2"]
+                      }
+                    }
+                  }
+                }
+                """;
+
+    String createCollectionWithDenyIndexing =
+        """
+                {
+                  "createCollection": {
+                    "name": "simple_collection_deny_indexing",
+                    "options" : {
+                      "vector" : {
+                        "size" : 5,
+                        "function" : "cosine"
+                      },
+                      "indexing" : {
+                        "deny" : ["field1", "field2"]
+                      }
+                    }
+                  }
+                }
+                """;
+
+    String createCollectionWithIndexingError =
+        """
+                {
+                  "createCollection": {
+                    "name": "simple_collection_error1",
+                    "options" : {
+                      "vector" : {
+                        "size" : 5,
+                        "function" : "cosine"
+                      },
+                      "indexing" : {
+                        "allow" : ["field1", "field1"]
+                      }
+                    }
+                  }
+                }
+                """;
+
+    String createCollectionWithIndexingError2 =
+        """
+                {
+                  "createCollection": {
+                    "name": "simple_collection_error2",
+                    "options" : {
+                      "vector" : {
+                        "size" : 5,
+                        "function" : "cosine"
+                      },
+                      "indexing" : {
+                        "allow" : ["field1", "field2"],
+                        "deny" : ["field1", "field2"]
```

Comment:
Can we add tests for pattern matching (for example `address.city`) and `_id` fields?

Reply:
The PR currently covers only adding fields. When we do PR for handling fields for indexing in Shredder will add those.